nifi-commons/nifi-security-utils/src/main/java/org/apache/nifi/security/util/EncryptionMethod.java

```diff
@@ -26,37 +26,43 @@
-    SHA_128AES("PBEWITHSHAAND128BITAES-CBC-BC", "BC", true),
-    SHA_192AES("PBEWITHSHAAND192BITAES-CBC-BC", "BC", true),
-    SHA_256AES("PBEWITHSHAAND256BITAES-CBC-BC", "BC", true),
-    SHA_40RC2("PBEWITHSHAAND40BITRC2-CBC", "BC", true),
-    SHA_128RC2("PBEWITHSHAAND128BITRC2-CBC", "BC", true),
-    SHA_40RC4("PBEWITHSHAAND40BITRC4", "BC", true),
-    SHA_128RC4("PBEWITHSHAAND128BITRC4", "BC", true),
-    SHA256_128AES("PBEWITHSHA256AND128BITAES-CBC-BC", "BC", true),
-    SHA256_192AES("PBEWITHSHA256AND192BITAES-CBC-BC", "BC", true),
-    SHA256_256AES("PBEWITHSHA256AND256BITAES-CBC-BC", "BC", true),
-    SHA_2KEYTRIPLEDES("PBEWITHSHAAND2-KEYTRIPLEDES-CBC", "BC", true),
-    SHA_3KEYTRIPLEDES("PBEWITHSHAAND3-KEYTRIPLEDES-CBC", "BC", true),
-    SHA_TWOFISH("PBEWITHSHAANDTWOFISH-CBC", "BC", true),
-    PGP("PGP", "BC", false),
-    PGP_ASCII_ARMOR("PGP-ASCII-ARMOR", "BC", false);
+    MD5_128AES("PBEWITHMD5AND128BITAES-CBC-OPENSSL", "BC", false, false),
+    MD5_192AES("PBEWITHMD5AND192BITAES-CBC-OPENSSL", "BC", true, false),
+    MD5_256AES("PBEWITHMD5AND256BITAES-CBC-OPENSSL", "BC", true, false),
+    MD5_DES("PBEWITHMD5ANDDES", "BC", false, false),
+    MD5_RC2("PBEWITHMD5ANDRC2", "BC", false, false),
+    SHA1_RC2("PBEWITHSHA1ANDRC2", "BC", false, false),
+    SHA1_DES("PBEWITHSHA1ANDDES", "BC", false, false),
+    SHA_128AES("PBEWITHSHAAND128BITAES-CBC-BC", "BC", false, false),
+    SHA_192AES("PBEWITHSHAAND192BITAES-CBC-BC", "BC", true, false),
+    SHA_256AES("PBEWITHSHAAND256BITAES-CBC-BC", "BC", true, false),
+    SHA_40RC2("PBEWITHSHAAND40BITRC2-CBC", "BC", false, false),
+    SHA_128RC2("PBEWITHSHAAND128BITRC2-CBC", "BC", false, false),
+    SHA_40RC4("PBEWITHSHAAND40BITRC4", "BC", false, false),
+    SHA_128RC4("PBEWITHSHAAND128BITRC4", "BC", false, false),
+    SHA256_128AES("PBEWITHSHA256AND128BITAES-CBC-BC", "BC", false, false),
+    SHA256_192AES("PBEWITHSHA256AND192BITAES-CBC-BC", "BC", true, false),
+    SHA256_256AES("PBEWITHSHA256AND256BITAES-CBC-BC", "BC", true, false),
+    SHA_2KEYTRIPLEDES("PBEWITHSHAAND2-KEYTRIPLEDES-CBC", "BC", false, false),
+    SHA_3KEYTRIPLEDES("PBEWITHSHAAND3-KEYTRIPLEDES-CBC", "BC", false, false),
+    SHA_TWOFISH("PBEWITHSHAANDTWOFISH-CBC", "BC", false, false),
+    PGP("PGP", "BC", false, false),
+    PGP_ASCII_ARMOR("PGP-ASCII-ARMOR", "BC", false, false),
+    // New encryption methods which used keyed encryption
+    AES_CBC("AES/CBC/PKCS7Padding", "BC", false, true),
+    AES_CTR("AES/CTR/NoPadding", "BC", false, true),
+    AES_GCM("AES/GCM/NoPadding", "BC", false, true);
 
     private final String algorithm;
     private final String provider;
     private final boolean unlimitedStrength;
+    private final boolean compatibleWithStrongKDFs;
 
-    EncryptionMethod(String algorithm, String provider, boolean unlimitedStrength) {
+    EncryptionMethod(String algorithm, String provider, boolean unlimitedStrength, boolean compatibleWithStrongKDFs) {
```

Comment:
Can this be a straight override or do we need to keep the original method and deprecate it?